torchaudio/functional.py

```diff
@@ -1239,6 +1240,60 @@ def dcshift(
     return output_waveform
 
 
+def overdrive(
+        waveform: Tensor,
+        gain: float = 20,
+        colour: float = 20
+) -> Tensor:
+    r"""Apply a overdrive effect to the audio. Similar to SoX implementation.
+    This effect applies a non linear distortion to the audio signal.
+
+    Args:
+        waveform (Tensor): audio waveform of dimension of `(..., time)`
```

Comment:
Looking at the implementation, it seems that this function expects some shape of Tensor. Can you elaborate that here?

I cannot infer what is the shape you are referring in the later part of the code.

```
last_in = torch.zeros(waveform.shape[0], waveform.shape[1])
last_out = torch.zeros(waveform.shape[0], waveform.shape[1])
```

Reply:
I have updated the function to take any shape waveform given that last dimension is time